projects/components/src/dropdown/dropdown-focus-handler.directive.ts

```diff
@@ -0,0 +1,165 @@
+import { Subscription } from 'rxjs';
+import { SubscriptionTracker } from '../common/subscription';
+import { DropdownFocusHandlerService } from './dropdown-focus-handler.service';
+import { DropdownComponent } from './dropdown.component';
+
+const BUTTON_NODE_NAME = 'BUTTON';
+
+/**
+ * Arrow keys directions
+ */
+export enum Direction {
+    UP = 'up',
+    DOWN = 'down',
+    LEFT = 'left',
+    RIGHT = 'right',
+}
+
+/**
+ * Object wrapping the focusable HTML elements of dropdown menu with the neighbors in 4 directions of each menu item.
+ */
+export interface MenuItem {
+    element: HTMLElement;
+    up?: MenuItem;
+    down?: MenuItem;
+    left?: MenuItem;
+    right?: MenuItem;
+    closeMenu?: (event: Event) => void;
+}
+
+/**
+ * Added on to vcd-dropdown component to vertically link menu items of the dropdown on which this directive is added. It also, then links
+ * the vertically linked menu items horizontally to their menu trigger. It then uses the {@link DropdownFocusHandlerService} to move the
+ * DOM focus between the menu items.
+ *
+ * @Example:
+ * <vcd-dropdown
+ *        vcdDropdownFocusHandler
+ *        [items]="contextualActions"
+ *        [onItemClickedCb]="runActionHandler.bind(this)"
+ *        [isItemDisabledCb]="isActionDisabled.bind(this)"
+ * ></vcd-dropdown>
+ */
+@Directive({
+    selector: 'vcd-dropdown[vcdDropdownFocusHandler]',
+})
+export class DropdownFocusHandlerDirective<T> implements AfterViewInit, OnDestroy {
+    constructor(
+        @Optional() @SkipSelf() private parentVcdDropdown: DropdownComponent<T>,
+        @Optional() @SkipSelf() private parentFocusHandler: DropdownFocusHandlerDirective<T>,
+        @Host() private hostVcdDropdown: DropdownComponent<T>,
+        private focusHandlerService: DropdownFocusHandlerService
+    ) {}
+
+    /**
+     * List of focusable menu items with their neighbors in 4 directions.
+     */
+    menuItems: MenuItem[];
+
+    private menuTrigger: MenuItem;
+    private isRootDropdown = !this.parentVcdDropdown;
+    private timeoutId: number;
+    private subscriptionTracker = new SubscriptionTracker(this);
+
+    /**
+     * After a dropdown menu is opened, it creates {@link MenuItem} for each of the menu items along with their trigger menu item and links
+     * them. It also then moves the focus to first item in the opened menu list.
+     */
+    ngAfterViewInit(): void {
+        this.subscriptionTracker.subscribe(this.hostVcdDropdown.dropdownMenuUpdated, (updated) => {
+            if (!updated) {
+                return;
+            }
+            this.createMenuTrigger();
+            this.linkMenuItems();
+            this.registerRootMenuContainer();
+            this.moveFocusToFirstItem();
+        });
+    }
+
+    ngOnDestroy(): void {}
+
+    private registerRootMenuContainer(): void {
+        if (this.isRootDropdown) {
+            const rootMenuContainer = this.hostVcdDropdown._clrDropdownMenuEl;
```

Comment:
Instead of relying that `this.hostVcdDropdown._clrDropdownMenuEl` is set, would it be possible to pass the `rootMenuContainer` as an argument to this method and call it from the dropdown component when _clrDropdownMenuEl is set? Same for the place that you use `_dropdownTriggerEl`? Just relying that some internal property within this.hostVcdDropdown is set feels strange.